app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchema.java

```diff
@@ -0,0 +1,168 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIgnore;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMappingException;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshaller;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.json.JsonUtils;
+import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
+import org.sagebionetworks.bridge.models.upload.UploadSchema;
+import org.sagebionetworks.bridge.validators.Validate;
+
+/**
+ * The DynamoDB implementation of UploadSchema. This is a mutable class with getters and setters so that it can work
+ * with the DynamoDB mapper.
+ */
+@DynamoDBTable(tableName = "UploadSchema")
+public class DynamoUploadSchema implements DynamoTable, UploadSchema {
+    private List<UploadFieldDefinition> fieldDefList;
+    private String name;
+    private int rev;
+    private String schemaId;
+    private String studyId;
+
+    /** {@inheritDoc} */
+    @DynamoDBMarshalling(marshallerClass = FieldDefinitionListMarshaller.class)
+    @Override
+    public List<UploadFieldDefinition> getFieldDefinitions() {
+        return fieldDefList;
+    }
+
+    /** @see org.sagebionetworks.bridge.models.upload.UploadSchema#getFieldDefinitions */
+    public void setFieldDefinitions(List<UploadFieldDefinition> fieldDefList) {
+        this.fieldDefList = ImmutableList.copyOf(fieldDefList);
+    }
+
+    /**
+     * This is the DynamoDB key. It is used by the DynamoDB mapper. This should not be used directly. The key format is
+     * "[studyID]$[schemaID]". The schema ID may contain dollar signs. The study ID may not. Since the key is created
+     * from the study ID and schema ID, this will throw an InvalidEntityException if either one is blank.
+     */
+    @DynamoDBHashKey
+    @JsonIgnore
+    public String getKey() throws InvalidEntityException {
+        if (Strings.isNullOrEmpty(studyId)) {
```

Comment:
If we use Guava's checkArgument, we could combine the check and exception throwing into one line of code.